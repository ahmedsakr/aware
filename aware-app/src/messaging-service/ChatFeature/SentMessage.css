.sentMessage {
    margin-top: 10px;
    display: flex;
    margin-right: 5px;
}

.sentMessage-profile {
    border: 2px solid rgb(54, 56, 58);
    border-radius: 5px;
    text-align: center;
    padding: 0px;
    margin-left: 5px;
}

.sentMessage-profile-img {
    max-width: 30%;
}

.sentMessage-profile-name {
    background-color: #bac1b8;
    text-align: center;
    margin: 0px;
}

.sentMessage-text {
    background-color: #769fb6;
    border-radius: 10px;
    padding: 5px;
    margin-left: 5px;
    text-align: right;
}

.sentMessage-timestamp {
    padding: 0px;
    text-align: right;
    font-size: 0.6em;
}
