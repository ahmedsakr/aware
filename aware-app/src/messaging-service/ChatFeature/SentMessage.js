import React, { Component } from 'react';
import './SentMessage.css';

class Message extends Component {

  render() {
    const { student, avatar, message, timestamp } = this.props;

    return (
      <div class="message">
        <div class="col-sm-3 message-timestamp">
          {timestamp}
        </div>
        <div class="col-sm-7 message-text">
          {message}
        </div>
        <div class="col-sm-2 message-profile">
          <img class="message-profile-img" src={process.env.PUBLIC_URL + avatar} alt={student} />
          <p class="message-profile-name">{student}</p>
        </div>
      </div>
    );
  }
}

export default Message;
