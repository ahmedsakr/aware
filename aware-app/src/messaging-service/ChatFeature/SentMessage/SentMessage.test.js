import React from 'react';
import { shallow } from 'enzyme';

import SentMessage from './SentMessage';

describe('SentMessage', () => {
  it('should render correctly in "debug" mode', () => {

    const component = shallow(<SentMessage debug />);
  
    expect(component).toMatchSnapshot();
  });
});