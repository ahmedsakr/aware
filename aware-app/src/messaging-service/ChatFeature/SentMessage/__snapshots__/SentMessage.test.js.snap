// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SentMessage should render correctly in "debug" mode 1`] = `
<div
  class="sentMessage"
>
  <div
    class="col-sm-3 sentMessage-timestamp"
  />
  <div
    class="col-sm-7 sentMessage-text"
  />
  <div
    class="col-sm-2 sentMessage-profile"
  >
    <img
      class="sentMessage-profile-img"
      src="undefined"
    />
    <p
      class="sentMessage-profile-name"
    />
  </div>
</div>
`;
